.github/workflows/macos.yml

```diff
@@ -46,7 +43,7 @@ jobs:
 
       - name: Install dependencies from Brew
         run: |
-          brew install --formula ninja cmake ccache icu4c boost gflags glog libevent
+          brew install --formula ninja flex bison cmake ccache icu4c boost gflags glog libevent
```

Comment:
maybe move this to _setup-macos.sh ? This way we need make changes only there and not have to make changes to nightly-conda too.

Reply:
Want to make sure we are using the same command in CI as we illustrates in the README about how to build from source: https://github.com/facebookresearch/torcharrow#install-dependencies

This helps CI to capture more potential issues for community developer experience.